internal/packets.go

```diff
@@ -151,22 +151,22 @@ func PacketChannelPartyUpdateInfo(partyID, playerID, job, level int32, name stri
 	return p
 }
 
-func PacketRateOperation(mode byte, rate int16) mpacket.Packet {
+func PacketRateOperation(mode byte, rate string) mpacket.Packet {
 	p := mpacket.CreateInternal(opcode.ChangeRate)
 	p.WriteByte(mode)
-	p.WriteInt16(rate)
+	p.WriteString(rate)
```

Comment:
Is it because there is no Write/ReadFloat32/64 method? If so you can implement one. Can't remember the endianess we need for the game but it would be something like:
```
b := *(*uint32)(unsafe.Pointer(&value))
*p = append(byte(b), byte(b >> 8), byte(b >> 16), byte(b >> 24))
```
then do the inverse operation for the read